src/model/element-base.ts

```diff
@@ -69,23 +78,50 @@ export class Slot {
   }
 }
 
+export interface Demo {
+  desc?: string;
+  path: string;
+}
+
+export interface ElementBaseInit {
+  description: string;
+  summary: string;
+  demos?: Demo[];
+  events?: Event[];
+  sourceRange: SourceRange|undefined;
+  properties?: Property[];
+  attributes?: Attribute[];
+  methods?: Method[];
+  astNode: any;
+  warnings?: Warning[];
+  slots?: Slot[];
+  privacy: Privacy;
+  jsdoc?: jsdoc.Annotation;
+  superClass?: ScannedReference;
+  mixins?: ScannedReference[];
+}
+
 /**
  * Base class for Element and ElementMixin.
  */
 export abstract class ElementBase implements Feature {
-  properties: Property[] = [];
-  attributes: Attribute[] = [];
-  methods: Method[] = [];
-  description = '';
-  summary = '';
-  demos: {desc?: string; path: string}[] = [];
-  events: Event[] = [];
-  sourceRange: SourceRange;
-  jsdoc?: jsdoc.Annotation;
-  astNode: estree.Node|null;
-  abstract kinds: Set<string>;
-  warnings: Warning[] = [];
+  readonly properties: Property[] = [];
+  readonly attributes: Attribute[] = [];
+  readonly methods: Method[] = [];
+  description: string;
+  readonly summary: string;
+  readonly demos: Demo[] = [];
+  readonly events: Event[] = [];
+  readonly sourceRange: SourceRange|undefined;
```

Comment:
I feel like `foo?: bar` is often worse than `foo: bar|undefined` in these sorts of cases, because we really want to be sure that we haven't forgotten about the field. It also means that we'll have more consistent object shapes. 

For class members though it's currently exactly the same, but in the future we may be able to get either typescript or tslint to warn about unassigned readonly properties.